client/my-sites/patterns/hooks/use-record-patterns-event.tsx

```diff
@@ -0,0 +1,20 @@
+import { recordTracksEvent } from 'calypso/lib/analytics/tracks';
+import { usePatternsContext } from 'calypso/my-sites/patterns/context';
+import { getTracksPatternType } from 'calypso/my-sites/patterns/lib/get-tracks-pattern-type';
+
+export const useRecordPatternsEvent = () => {
+	const { category, patternTypeFilter } = usePatternsContext();
+
+	return {
+		recordPatternsEvent: (
+			tracksEventName: string,
+			params: Record< string, string | number | undefined > = {}
+		) => {
+			recordTracksEvent( tracksEventName, {
+				category,
```

Comment:
```suggestion
				category: category || undefined,
```

Category may be empty in the search result page. This avoids reporting an empty `category` property.

Reply:
Hm, maybe I am missing something but it seems it doesn't make any sense, unless we provide `''` (empty string).
Am I wrong?

Reply:
By "empty", do you mean that - to avoid tracking "empty string" (`""`) we should set `undefined`?

Reply:
>avoid tracking "empty string" ("") we should set undefined

Correct!